Tests/CSharp/CodeRefactorings/ContractEnsuresNotNullReturnTests.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using NUnit.Framework;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis;
+using System.Collections.Immutable;
+using RefactoringEssentials.CSharp.CodeRefactorings;
+using System.Diagnostics.Contracts;
+
+namespace RefactoringEssentials.Tests.CSharp.CodeRefactorings
+{
+    [TestFixture]
+    public class ContractEnsuresNotNullReturnTests : CSharpCodeRefactoringTestBase
+    {
+        public int? play()
+        {
+            Contract.Ensures(Contract.Result<int?>() != null);
+            return null;
+        }
+
+        public ContractEnsuresNotNullReturnCodeRefactoringProvider play2()
+        {
+            return null;
+        }
+
+        [Test]
+        public void ValueTypeReturnType()
+        {
+            TestWrongContext<ContractEnsuresNotNullReturnCodeRefactoringProvider>(
+                @"class Test
+            {
+                public $int Foo() 
+                { 
+                }
+            }");
+        }
+
+        [Test]
+        public void NullableValueTypeLocalVariable()
+        {
+            TestWrongContext<ContractEnsuresNotNullReturnCodeRefactoringProvider>(
+                @"class Test
+            {
+                public void Foo() 
+                { 
+                    $int? foo;
+                }
+            }");
+        }
+
+        [Test]
+        // Not sure why anyone would want this when can just change the return type to be non nullable. Maybe you have to implement an interface you don't control or something strange like that.
+        public void NullableValueTypeReturnType()
+        {
+            Test<ContractEnsuresNotNullReturnCodeRefactoringProvider>(
+                @"class Test
+{
+    $public int? Foo() 
+    { 
+    }
+}", @"class Test
+{
+    public int? Foo() 
+    {
+        Contract.Ensures(Contract.Result<int?>() != null);
+    }
+}");
+        }
+
+        [Test]
+        public void ObjectReturnType()
+        {
+            Test<ContractEnsuresNotNullReturnCodeRefactoringProvider>(
+                @"class Test
+{
+    $public Cedd Foo() 
+    { 
+    }
+}", @"class Test
+{
+    public Cedd Foo() 
+    {
+        Contract.Ensures(Contract.Result<Cedd>() != null);
+    }
+}");
+        }
+
```

Comment:
My comment was more about the unit tests and the cases they check. It's clear that a not-null contract doesn't make sense for a void method, but then there should be a unit test ensuring that this refactoring is not suggested for void methods.

BTW: You don't have to use `SpecializedCodeRefactoringProvider` all the time. If you want to cover different syntax node types you always have the option to derive the refactoring directly from `CodeRefactoringProvider` and then check the types yourself.

Covering property getters and indexers would be great, lambdas and delegates are optional.

And of course: Thank you for the work so far!